src/geometricconvolutions/geometric.py

```diff
@@ -743,11 +742,8 @@ def times_group_element(self, gg):
         else:
             # applying the rotation to tensors is essentially multiplying each index, which we can think of as a
             # vector, by the group action. The image pixels have already been rotated.
-            firstletters  = "abcdefghijklm"
-            secondletters = "nopqrstuvwxyz"
-            einstr = "".join([firstletters[i] for i in range(self.D)])
-            einstr += "".join([firstletters[i + self.D] for i in range(self.k)]) + ","
-            einstr += ",".join([secondletters[i] + firstletters[i + self.D] for i in range(self.k)])
+            einstr = LETTERS[:len(self.shape())] + ','
+            einstr += ",".join([LETTERS[i+13] + LETTERS[i + len(self.image_shape())] for i in range(self.k)])
```

Comment:
For some reason, it didn't like when the capitals letters are in there. I am not sure what was up, capital letters should be treated as distinct characters from the lowercase letters, but there order gets messed up or something.